main/admission.hh

```diff
@@ -7,6 +7,6 @@
 void admission_itemname_hash(Str& ih,const Str& name,const Str& location,const Str& time, const Str& price_class); //converts a name to the item~hash form.
 void admission_parse_hash_name(Str& name,const Str& ih);
 void admission_parse_hash_ltime_hash(Str& hout,const Str& ih);
-const char* admission_filteredname(const Str& item_name);
+const char* admission_filteredname(const std::string& item_name);  // REFACTOR: Changed from Str& to std::string& for modernization
```

Comment:
I think this refactor can be reverted completely. This would only make sense if we replace all Str. Not just one function. So please revert this refactor in admission.hh anc cc